usaspending_api/etl/tests/integration/test_load_table_to_delta.py

```diff
@@ -0,0 +1,87 @@
+"""Automated Integration Tests for the lifecycle of Delta Lake tables
+
+NOTE: Uses Pytest Fixtures from immediate parent conftest.py: usaspending_api/etl/tests/conftest.py
+"""
+from datetime import datetime
+
+from model_bakery import baker
+from pyspark.sql import SparkSession
+from pytest import mark
+from django.core.management import call_command
+
+from usaspending_api.etl.management.commands.create_delta_table import TABLE_SPEC
+
+
+def equal_datasets(ds1, ds2):
+    """Helper function to compare the two datasets. Note the column types of ds1 will be used to cast columns in ds2."""
+    datasets_match = True
+    for i, ds1_row in enumerate(ds1):
+        for k, val_1 in ds1_row.items():
+            val_1_type = type(val_1)
+            val_2 = ds2[i][k]
+            if not isinstance(val_1, type(None)) and not isinstance(val_1, type(val_2)):
+                val_2 = val_1_type(val_2)
```

Comment:
Why perform a cast here? Don't we want to catch types that don't match?

Reply:
Yeah we need to think about what scenarios may cause a failure. Maybe there are a handful of types we treat as "synonymous" and tolerate casting, but others we don't? ... e.g. numeric and decimal ... but not decimal and int for instance?

Reply:
What type of mismatches are you seeing here? 
If these mismatches are caused by the casting in the ETL done by `custom_schema`, would it make sense to use the `custom_schema` property from the `TABLE_SPEC` to target specific casts?